crates/server/src/sse/handlers.rs

```diff
@@ -0,0 +1,342 @@
+                let mut inner = session.inner.write().await;
+                let mut invalid = Vec::new();
+
+                for ctx in &ctxs.context_ids {
+                    if !inner.subscriptions.remove(ctx) {
+                        invalid.push(*ctx);
+                    }
+                }
+                inner.touch();
+
+                // Persist to store
+                let persisted = inner.to_persisted();
+                drop(inner);
+                drop(sessions);
+
+                let mut store = state.store.clone();
+                if let Err(err) = save_session(&mut store, session_id, &persisted) {
+                    error!(%session_id, %err, "Failed to persist session after unsubscribe");
+                }
+
+                if !invalid.is_empty() {
+                    (
+                        StatusCode::BAD_REQUEST,
+                        Json(Response {
+                            body: ResponseBody::Error(ResponseBodyError::HandlerError(
+                                "Some context IDs were not subscribed".into(),
+                            )),
+                        }),
+                    )
+                } else {
+                    (
+                        StatusCode::OK,
+                        Json(Response {
+                            body: ResponseBody::Result(serde_json::json!({
+                                "status": "unsubscribed",
+                                "contexts": ctxs.context_ids,
+                            })),
+                        }),
+                    )
+                }
+            } else {
+                (
+                    StatusCode::NOT_FOUND,
+                    Json(Response {
+                        body: ResponseBody::Error(ResponseBodyError::HandlerError(
+                            "Session not found. Please reconnect to SSE endpoint first.".into(),
+                        )),
+                    }),
+                )
+            }
+        }
+        Err(err) => (
+            StatusCode::BAD_REQUEST,
+            Json(Response {
+                body: ResponseBody::Error(ResponseBodyError::ServerError(
+                    ServerResponseError::ParseError(err.to_string()),
+                )),
+            }),
+        ),
+    }
+}
+
+/// Handle SSE connection establishment
+#[expect(
+    clippy::too_many_lines,
+    reason = "Complex handler with multiple reconnection paths"
+)]
+pub async fn sse_handler(
+    Extension(state): Extension<Arc<ServiceState>>,
+    request: AxumRequest,
+) -> Sse<impl Stream<Item = Result<Event, Infallible>>> {
+    let headers = request.headers();
+
+    // Check for Last-Event-ID header for reconnection
+    let last_event_id = headers
+        .get("Last-Event-ID")
+        .and_then(|v| v.to_str().ok())
+        .and_then(|s| s.split('-').next())
+        .and_then(|id| id.parse::<u64>().ok());
+
+    let (commands_sender, commands_receiver) = mpsc::channel::<Command>(32);
```

Comment:
Why this buffer size exactly? nevertheless, I think it's better to use a const or put that value into the `SseConfig`.